File: Shared/Resources/oddevensort.bundle/oddevensort.py
def sort(arr):
  sorted = False
  while not sorted:
    sorted = True
    for i in range(1, len(arr) - 1, 2):
      if arr[i] > arr[i + 1]:
        arr[i], arr[i + 1] = arr[i + 1], arr[i]
        sorted = False
    for i in range(0, len(arr) - 1, 2):
      if arr[i] > arr[i + 1]:
        arr[i], arr[i + 1] = arr[i + 1], arr[i]
        sorted = False

if __name__ == "__main__":
  array = [35, 95, 74, 71, 72, 30, 96, 53, 9, 0]
  sort(array)
  print(array)

File: Shared/Resources/oddevensort.bundle/oddevensort.rb
def sort(arr)
  n = arr.length()
  sorted = false
  while !sorted
    sorted = true
    i = 1
    while i < n - 1
      if arr[i] > arr[i + 1]
        arr[i], arr[i + 1] = arr[i + 1], arr[i]
        sorted = false
      end
      i += 2
    end
    i = 0
    while i < n - 1
      if arr[i] > arr[i + 1]
        arr[i], arr[i + 1] = arr[i + 1], arr[i]
        sorted = false
      end
      i += 2
    end
  end
end

array = [0, 39, 21, 62, 91, 77, 14, 23,
  90, 69, 51, 81, 68, 83, 32, 56]
sort(array)
p array